libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -875,8 +868,8 @@ func mirroredWSL2Workaround(config configuration, ipv iptables.IPVersion) error
 //     running - no workaround is needed, the normal DNAT/masquerading works.
 //   - and, the host Linux appears to be running under Windows WSL2 with mirrored
 //     mode networking.
-func insertMirroredWSL2Rule(config configuration) bool {
-	if !config.EnableUserlandProxy || config.UserlandProxyPath == "" {
+func insertMirroredWSL2Rule(hairpin bool) bool {
```

Comment:
nit: the function name suggests that it actually **inserts** the rule and not checks if it should be inserted.
Not related to this PR, but in case someone wants to change it in a follow up:

```suggestion
func shouldInsertMirroredWSL2Rule(hairpin bool) bool {
```

Reply:
Fair point! Thank you ... I'll fix it.